packages/blade/src/components/Dropdown/__tests__/Dropdown.web.test.tsx

```diff
@@ -520,6 +520,81 @@ describe('<Dropdown />', () => {
     Object.defineProperty(HTMLElement.prototype, 'clientWidth', { configurable: true, value: 0 });
   });
 
+  // https://github.com/razorpay/blade/issues/1721
+  it('should handle controlled props & disabled options with multi select', async () => {
+    const ControlledDropdown = (): React.ReactElement => {
+      const [currentSelection, setCurrentSelection] = React.useState<string[]>([]);
+      const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+      return (
+        <>
+          <Button
+            onClick={() => {
+              if (!currentSelection.includes('bangalore')) {
+                setCurrentSelection([...currentSelection, 'bangalore']);
+              }
+            }}
+          >
+            Select Bangalore
+          </Button>
+          <Button onClick={() => setIsDropdownOpen(!isDropdownOpen)}>Toggle Dropdown</Button>
+
+          <Dropdown
+            isOpen={isDropdownOpen}
+            onOpenChange={setIsDropdownOpen}
+            selectionType="multiple"
+          >
+            <SelectInput
+              label="Select City"
+              value={currentSelection}
+              onChange={(args) => {
+                if (args) {
+                  setCurrentSelection(args.values);
+                }
+              }}
+            />
+            <DropdownOverlay>
+              <ActionList>
+                <ActionListItem title="Delhi" value="delhi" isDisabled />
```

Comment:
This test won't exactly test that github issue. This case is anyway working fine - https://codesandbox.io/s/snowy-water-687dxg?file=/App.tsx. It's just sandpack sometimes acts up. Things work fine when you refresh page.

Their issue is dynamic ActionListItem coming from options and then some properties are changing based on selection of items. A bit difficult to replicate that in tests.